src/sprites/Elevator.lua

```diff
@@ -72,118 +66,225 @@ function Elevator:collisionResponse(_)
+    elseif key == KEYNAMES.Left then
+      return -self.speed
+    end
+  else
+    -- Vertical orientation, return positive if Down, negative if Up
 
---- Get remaining movement based on direction and displacement
-local function getMovementRemaining(self, key)
-  -- Get inverted direction boolean
-  local isInverseDirection = isInverseDirection(key)
-
-  -- Remaining displacement - either towards displacementEnd or displacementStart if inverse direction
-  local displacementRemaining = isInverseDirection 
-    and self.displacement - self.displacementStart
-    or self.displacementEnd - self.displacement
-  
-  -- Calculate movement as scalar value, invert speed if inverse direction
+    if key == KEYNAMES.Down then
+      return self.speed
+    elseif key == KEYNAMES.Up then
+      return -self.speed
+    end
+  end
 
-  return math.min(displacementRemaining, self.speed) * (isInverseDirection and -1 or 1)
-end
+  -- If key is not a correct activation key, return 0
 
--- Public class Methods
+  return 0
+end
 
---- Sets movement to be executed in the next update() call using vector.
---- *param* key - the player input key direction (KEYNAMES)
---- *returns* whether an activation occurred based on key press.
-function Elevator:activate(key)
-  local movement = 0
-
-  if (key == KEYNAMES.Left or key == KEYNAMES.Right)
-  and self.fields.orientation == ORIENTATION.Horizontal then
-    -- Horizontal movement - get distance remaining
-    movement = getMovementRemaining(self, key)
-
-  elseif (key == KEYNAMES.Down or key == KEYNAMES.Up)
-  and self.fields.orientation == ORIENTATION.Vertical then
-    -- Vertical movement - get distance remaining
-    movement = getMovementRemaining(self, key)
+--- Get remaining movement based on direction and displacement
+local function getMovementRemaining(self, movement)
+  if movement < 0 then
+    return math.max(-self.displacement, movement)
+  elseif movement > 0 then
+    return math.min(self.displacementEnd - self.displacement, movement)
+  else
+    return 0
   end
-  
-  -- Set movement update scalar
-  self.movement = movement
+end
 
-  self.displacement += self.movement * _G.delta_time
+--- Checks collision for frame, also checking if child collides. Returns a partial movement for itself
+--- if elevator or child collides with another object.
+local function checkIfCollides(self, idealX, idealY, spriteToCheck)
+  local spriteToCheck = spriteToCheck or self
+  local idealSpriteToCheckX = idealX - self.x + spriteToCheck.x
+  local idealSpriteToCheckY = idealY - self.y + spriteToCheck.y
 
-  -- If close to start or end, adjust displacement & cancel movement
-  if self.displacement - self.displacementStart < 1 or
-    self.displacementEnd - self.displacement < 1 then
-      self:displacementAdjustToTile()
-      self.movement = 0
+  -- Check if elevator collides
+
+  local actualSpriteToCheckX, actualSpriteToCheckY, collisions = spriteToCheck:checkCollisions(idealSpriteToCheckX, idealSpriteToCheckY)
```

Comment:
it's too bad we have to run collision checks again, ostensibly the sprite is already making those during any RB updates. Probably isn't going to slow us down in practice too much knowing the game, just a thought

Reply:
This is a very valid point. For now I don't have a better idea... the PD collision system can't handle two moving sprites moving into each other, so we'd have to implement our own collision resolving potentially. Agree that RB can be the source of this improvement.